compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/gestures/TapGesturesDetector.skiko.kt

```diff
@@ -194,7 +215,7 @@ private suspend fun AwaitPointerEventScope.awaitReleaseOrCancelled(
             it.isOutOfBounds(size, Size.Zero)
         }
 
-        if (cancelled) return null
+        if (cancelled || cancelIf(event)) return null
 
         event = event.takeIf {
```

Comment:
The logic was taken from the original (from androidx) long click processing (after onLongClick event, it waits for up event and consumes all other events until up`consumeUntilUp()`): https://github.com/JetBrains/androidx/blob/fecbee52ee7a0d5c306f55ce445e14b9fd8e93b3/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/TapGestureDetector.kt#L120

Your question makes sense to me. I'm not 100% sure, but I think the idea is that while user keeps the button pressed (so long click was not released), no other components should receive events for that long-clicked component. It's somewhat similar to drag: I pressed the button and move the pointer, so all events kind of belong to the clicked component. 

An example why it's worth to consume all events before UP:
`BoxA(onDragUsingRightMouseButton = {}) {
     BoxB(onLongClick = {}) 
 }`

If I long-click using left button on a BoxB, and then without releasing the left button, I press down the right button and keep moving the pointer, the drag of BoxA would start (if move events on BoxA are not consumed).

Do you think it's worth to consume events for such cases?

Reply:
```
// After long click was detected and dispatched, we should wait for the pointer button release.
    // In this case the event has been already dispatched, therefore, the release event is not required to
    // have the matching keyboard modifiers.
    val longClickReleaseFilter: (PointerEvent) -> Boolean = {
        matcher.matches(it)
    }
    ```

Reply:
The case with drag and long click seems make sense. For example, user wants to detect a drag to start moving, and detect a long click to change selection mode. And they can't do that if we consume events waiting for a long click.

Usually, it is better to stick to the original behaviour, but it seems the original behaviour has a flaw.

Reply:
Regarding the flaw. Not sure I understood:

> For example, user wants to detect a drag to start moving, and detect a long click to change selection mode. And they can't do that if we consume events waiting for a long click.

Do you describe the flaw in that example?

According to my understanding the following will happen:

1) User presses the left mouse button on a Box 
2) X ms seconds pass so onLongClick is dispatched
3) User keep the left mouse button pressed.

Since the left button is still pressed, there is no way user can initiate drag untill left button is released. 
So consuming all events before release, doesn't break anything here. Right?

Reply:
> Since the left button is still pressed, there is no way user can initiate drag untill left button is released.

We need to start drag before the long click and cancel the future long click. Imagine a file manager, where we can drag file immediately on a drag, and want to do something special on a long click (for example, enable multiple selection - we apply long click, and just move mouse down to select next files).